projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -542,29 +558,65 @@ export class IgxOverlayService implements OnDestroy {
         }
     }
 
-    private getOverlayInfo(component: any, moduleRef?: { injector: Injector, componentFactoryResolver: ComponentFactoryResolver }): OverlayInfo {
+    private getUserOverlaySettings(
+      viewContainerRefOrSettings: ViewContainerRef | OverlaySettings | undefined,
+      moduleRefOrSettings?: { injector: Injector, componentFactoryResolver: ComponentFactoryResolver } | OverlaySettings): OverlaySettings | undefined {
+        let result: OverlaySettings | undefined;
+        if (viewContainerRefOrSettings && !(viewContainerRefOrSettings instanceof ViewContainerRef)) {
+            result = viewContainerRefOrSettings;
+            return result;
+        }
+        if (moduleRefOrSettings && !('injector' in moduleRefOrSettings && 'componentFactoryResolver' in moduleRefOrSettings)) {
+            result = moduleRefOrSettings;
+        }
+        return result;
+    }
+
+
+    private getUserViewContainerOrModuleRef(
+        viewContainerRefOrSettings: ViewContainerRef | OverlaySettings | undefined,
+        moduleRefOrSettings?: { injector: Injector, componentFactoryResolver: ComponentFactoryResolver } | OverlaySettings
+        ): ViewContainerRef | { injector: Injector, componentFactoryResolver: ComponentFactoryResolver } | undefined {
+          let result: ViewContainerRef | { injector: Injector, componentFactoryResolver: ComponentFactoryResolver } | undefined;
+          if (viewContainerRefOrSettings instanceof ViewContainerRef) {
+              result = viewContainerRefOrSettings;
+          }
+          if (moduleRefOrSettings && 'injector' in moduleRefOrSettings && 'componentFactoryResolver' in moduleRefOrSettings) {
+              result = moduleRefOrSettings;
+          }
+          return result;
+      }
+
+    private getOverlayInfo(
+        component: ElementRef | Type<any>,
+        viewContainerRef?: { injector: Injector, componentFactoryResolver: ComponentFactoryResolver } | ViewContainerRef): OverlayInfo {
```

Comment:
```suggestion
        viewContainerRef?: { injector: Injector, componentFactoryResolver: ComponentFactoryResolver } | ViewContainerRef): OverlayInfo | null {
```

As the `catch` will return `null`.